### PR TITLE
Fixing popover position for scenarios contextual actions

### PR DESCRIPTION
**Pull request by AbdoKousara**
**fixing the Scenarios items popover position using
headlessui/popover and usePopper**

### CODE REVIEW

File: client/src/containers/scenarios/item/component.tsx [hunk 1]
```diff
@@ -94,95 +112,103 @@ const ScenariosList: React.FC<ScenariosItemProps> = (props: ScenariosItemProps)
...
+                          ref={setPopperElement}
+                          style={styles.popper}
+                          {...attributes.popper}
+                          static
+                          className="z-10 w-56 bg-white rounded-md shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none"
+                        >
+                          <Transition
+                            show={open}
+                            as={Fragment}
+                            enter="transition ease-out duration-1000"
```
Comment (davidsingal) [inline client/src/containers/scenarios/item/component.tsx:140]: duration-100
```suggestion
                            enter="transition ease-out duration-100"
```

**Review by davidsingal — CHANGES_REQUESTED** [review]

File: client/src/containers/scenarios/item/component.tsx [hunk 2]
```diff
@@ -18,9 +20,25 @@ type ScenariosItemProps = {
 const DROPDOWN_BUTTON_CLASSNAME =
   'w-8 h-8 inline-flex items-center justify-center text-gray-900 rounded-full bg-transparent hover:text-green-800';
 const DROPDOWN_ITEM_CLASSNAME = 'block px-4 py-2 text-sm w-full text-left';
-const DROPDOWN_ITEM_ACTIVE_CLASSNAME = 'bg-gray-100 text-gray-900';
+const DROPDOWN_ITEM_ACTIVE_CLASSNAME = 'hover:bg-gray-100 hover:text-gray-900';
```
Comment (mluena) [inline client/src/containers/scenarios/item/component.tsx:23]: you could add this hover interaction to previous class and get rid off this one

File: client/src/containers/scenarios/item/component.tsx [hunk 3]
```diff
@@ -18,9 +20,25 @@ type ScenariosItemProps = {
...
 const ScenariosList: React.FC<ScenariosItemProps> = (props: ScenariosItemProps) => {
+  const [referenceElement, setReferenceElement] = useState(null);
+  const [popperElement, setPopperElement] = useState(null);
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement: 'left-start',
+    strategy: 'absolute',
+    modifiers: [
+      {
+        name: 'offset',
+
```
Comment (mluena) [inline client/src/containers/scenarios/item/component.tsx:34]:
```suggestion
```

File: client/src/containers/scenarios/item/component.tsx [hunk 4]
```diff
@@ -18,9 +20,25 @@ type ScenariosItemProps = {
...
   'w-8 h-8 inline-flex items-center justify-center text-gray-900 rounded-full bg-transparent hover:text-green-800';
 const DROPDOWN_ITEM_CLASSNAME = 'block px-4 py-2 text-sm w-full text-left';
-const DROPDOWN_ITEM_ACTIVE_CLASSNAME = 'bg-gray-100 text-gray-900';
+const DROPDOWN_ITEM_ACTIVE_CLASSNAME = 'hover:bg-gray-100 hover:text-gray-900';
 
 const ScenariosList: React.FC<ScenariosItemProps> = (props: ScenariosItemProps) => {
+  const [referenceElement, setReferenceElement] = useState(null);
+  const [popperElement, setPopperElement] = useState(null);
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement: 'left-start',
```
Comment (mluena) [inline client/src/containers/scenarios/item/component.tsx:29]:
```suggestion
    placement: 'right-start',
```

File: client/src/containers/scenarios/item/component.tsx [hunk 5]
```diff
@@ -18,9 +20,25 @@ type ScenariosItemProps = {
...
+    strategy: 'absolute',
+    modifiers: [
+      {
+        name: 'offset',
+
+        options: {
+          offset: [-110, -20],
```
Comment (mluena) [inline client/src/containers/scenarios/item/component.tsx:36]:
```suggestion
          offset: [-20, 20],
```

**Review by mluena — CHANGES_REQUESTED** [review]

**Pull request merged**